facades/cmc_facade.py

```diff
@@ -2,25 +2,79 @@
 import locale
 import json
 import os
+import logging
 from dotenv import load_dotenv
 
 load_dotenv()
 
 def get_btc_price():
-  url = 'https://pro-api.coinmarketcap.com/v2/cryptocurrency/quotes/latest'
-
-  parameters = {
-    'symbol': 'BTC'
-  }
-
-  headers = {
-    'Accepts': 'application/json',
-    'X-CMC_PRO_API_KEY': os.getenv('CMC_API_KEY')
-  }
-  try: 
-    response = requests.get(url, params=parameters, headers=headers)
-    response.raise_for_status() # Raise an HTTPError for bad responses
-    price = response.json()['data']['BTC'][0]['quote']['USD']['price']
-    return price
-  except requests.exceptions.RequestException as e:
-    raise Exception("Failed to fetch BTC price") from e
\ No newline at end of file
+    # Fetches current BTC price from CoinMarketCap API
+    # Returns: float price in USD
+    # Raises: CMCApiError and its subclasses
+
+    url = 'https://pro-api.coinmarketcap.com/v2/cryptocurrency/quotes/latest'
+
+    parameters = {
+        'symbol': 'BTC'
+        }
+
+    headers = {
+        'Accepts': 'application/json',
+        'X-CMC_PRO_API_KEY': os.getenv('CMC_API_KEY')
+        }
+
+    if not os.getenv('CMC_API_KEY'):
+        raise CMCApiError("CMC API Key not found in environment variables")
```

Comment:
This check should occur before you use it in headers. assign it to a vaiable and check for it. better yet, move this to `consts.py`. if we weren't doing this in const something like:
```python
if not cmc_api_key := os.getend('CMC_API_KEY'):
    raise ...
    
# use cmc_api_key in headers
```
is cleaner and good use of the walrus operator.

Reply:
TIL there was such a thing as Walrus operator. Clever Python stuff